autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -325,23 +297,60 @@ def score_with_y_pred_proba(self, y, y_pred_proba, eval_metric=None, metric_need
         else:
             return eval_metric(y, y_pred_proba)
 
-    def save(self, file_prefix="", directory=None, return_filename=False, verbose=True):
-        if directory is None:
-            directory = self.path
-        file_name = directory + file_prefix + self.model_file_name
-        save_pkl.save(path=file_name, object=self, verbose=verbose)
-        if return_filename:
-            return file_name
+    def save(self, path: str = None, verbose=True) -> str:
+        """
+        Saves the model to disk.
+
+        Parameters
+        ----------
+        path : str, default None
+            Path to the saved model, minus the file name.
+            This should generally be a directory path ending with a '/' character (or appropriate path separator value depending on OS).
+            If None, self.path is used.
+            The final model file is typically saved to path + self.model_file_name.
+        verbose : bool, default True
+            Whether to log the location of the saved file.
+
+        Returns
+        -------
+        path : str
+            Path to the saved model, minus the file name.
+            Use this value to load the model from disk via cls.load(path), cls being the class of the model object, such as model = RFModel.load(path)
+        """
+        if path is None:
+            path = self.path
+        file_path = path + self.model_file_name
+        save_pkl.save(path=file_path, object=self, verbose=verbose)
+        return path
 
     @classmethod
-    def load(cls, path, file_prefix="", reset_paths=False, verbose=True):
-        load_path = path + file_prefix + cls.model_file_name
-        if not reset_paths:
-            return load_pkl.load(path=load_path, verbose=verbose)
-        else:
-            obj = load_pkl.load(path=load_path, verbose=verbose)
-            obj.set_contexts(path)
-            return obj
+    def load(cls, path: str, reset_paths=True, verbose=True):
+        """
+        Loads the model from disk to memory.
+
+        Parameters
+        ----------
+        path : str
+            Path to the saved model, minus the file name.
+            This should generally be a directory path ending with a '/' character (or appropriate path separator value depending on OS).
+            The model file is typically located in path + cls.model_file_name.
+        reset_paths : bool, default True
+            Whether to reset the self.path value of the loaded model to be equal to path.
+            It is highly recommended to keep this value as True unless accessing the original self.path value is important.
```

Comment:
maybe clarify for example when might this be important?

Reply:
Can revisit on doc overhaul for 0.1